internal/provider/volume/resource_volume.go

```diff
@@ -131,7 +143,10 @@ func resourceVolumeImport(d *schema.ResourceData, obj *corev1.PersistentVolumeCl
 
 func resourceVolumeRefresh(ctx context.Context, d *schema.ResourceData, meta interface{}) resource.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-		c := meta.(*client.Client)
+		c, err := meta.(*config.Config).K8sClient()
```

Comment:
This now constructs a new client instance every time this function runs.
Keep in mind, this function will be run in a loop to watch the state of a volume (e.g. while it is being removed).
Instead the client should be handed into this function from outside, so it can be keep throughout multiple invocations.

Reply:
Changed to reuse `K8sClient`.